src/operators/multipliedoperator.jl

```diff
@@ -0,0 +1,56 @@
+"""
+    Structure representing a multiplied operator. Consists of
+        - An operator op (MPO, Hamiltonian, ...)
+        - An object f that gets multiplied with the operator (Number, function, ...) 
+"""
+struct MultipliedOperator{O,F}
+    op::O
+    f::F
+end
+
+"""
+    Structure representing a time-dependent operator. Consists of
+        - An operator op (MPO, Hamiltonian, ...)
+        - An function f that gives the time-dependence according to op(t) = f(t)*op
+"""
+const TimedOperator{O} = MultipliedOperator{O,<:Function}
+
+"""
+    Structure representing a time-independent operator that will be multiplied with a constant coefficient. Consists of
+        - An operator (MPO, Hamiltonian, ...)
+        - A number f that gets multiplied with the operator
+"""
+const UntimedOperator{O} = MultipliedOperator{O,<:Union{Real,One}}
+
+#constructors for (un)TimedOperator
+TimedOperator(x::O, f::F) where {F<:Function,O} = MultipliedOperator(x, f)
+UntimedOperator(x::O, c::C) where {C<:Union{Real,One},O} = MultipliedOperator(x, c)
```

Comment:
Do we need explicit constructors for Timed and Untimed? Can we not always use MultipliedOperator, which automatically infers if it is Timed or Untimed based on the input arguments?

Reply:
though they are not strictly necessary, I found these constructors handy and intuitive.